tests/block_test.go

```diff
@@ -24,28 +24,90 @@ package tests
 
 import (
 	"testing"
+
+	"github.com/kaiachain/kaia/common"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestBlockchain(t *testing.T) {
-	t.Parallel()
+// TestExecutionSpecState runs the state_test fixtures from execution-spec-tests.
+type ExecutionSpecBlockTestSuite struct {
+	suite.Suite
+	originalIsPrecompiledContractAddress func(common.Address, interface{}) bool
+}
+
+func (suite *ExecutionSpecBlockTestSuite) SetupSuite() {
+	suite.originalIsPrecompiledContractAddress = common.IsPrecompiledContractAddress
+	common.IsPrecompiledContractAddress = isPrecompiledContractAddressForEthTest
```

Comment:
(1) in the case of Cancun, there is 0xa precompiled contract, but it seems to be ignored. Is this okay?
(2) Is it possible to replace the `PrecompiledContractsByzantium` instead of replacing IsPrecompiledContractAddress?

Reply:
In `ActivePrecompiles`, 0x0a and 0x0b have been unconditionally added to the active precompiles so I ignored them. But this was wrong. Here needs to get correct active precompiles for tests without using `ActivePrecompiles`.

Reply:
@yoomee1313 
> (1) in the case of Cancun, there is 0xa precompiled contract, but it seems to be ignored. Is this okay?

As Shogo said, we shouldn't ignore 0x0a and 0x0b. I adopted Shogo's proposal.

> (2) Is it possible to replace the PrecompiledContractsByzantium instead of replacing IsPrecompiledContractAddress?

The function needs to be replaced because the current `IsPrecompiledContractAddress` sets the entire range of `lastPrecompiledContractAddressHex` to false.
https://github.com/kaiachain/kaia/blob/906d8c2428ffa8e8b5d3191224f6836f61268005/common/types.go#L360

Reply:
(2) means  overwriting the existing global variable (PrecompiledAddressByzantium) in the test setup section rather than make `IsPrecompiledContract function` to be a variable.
I understood the explanation and no need to change like I mentioned.